node/src/actors/session/handlers.rs

```diff
@@ -223,9 +223,23 @@ impl StreamHandler<BytesMut, Error> for Session {
                             .into_actor(self)
                             .map_err(|e, _, _| log::error!("Inventory request error: {}", e))
                             .and_then(move |item_responses, session, _| {
+                                let mut send_superblock_votes = false;
                                 for (i, item_response) in item_responses.into_iter().enumerate() {
                                     match item_response {
-                                        Ok(item) => send_inventory_item_msg(session, item),
+                                        Ok(item) => {
+                                            if let InventoryItem::Block(block) = &item {
+                                                if block.hash()
```

Comment:
Can we compare the checkpoints instead? 

```suggestion
                                                if block.block_header.beacon.checkpoint
```